include/zephyr/bluetooth/buf.h

```diff
@@ -164,6 +164,18 @@ static inline enum bt_buf_type bt_buf_get_type(struct net_buf *buf)
 		->type;
 }
 
+struct bt_buf_view_meta {
+	struct net_buf *parent;
+	/* saves the data pointers while the parent buffer is locked. */
+	struct net_buf_simple backup;
```

Comment:
```suggestion
	struct net_buf_simple_state backup;
```

Reply:
Also see `net_buf_simple_save()` and `net_buf_simple_restore()` for how the `data` pointer is recovered using an offset and the `__buf` pointer.

Reply:
I remember now. We actually modify the `__buf` pointer to lock the buffer headroom.

https://github.com/zephyrproject-rtos/zephyr/blob/2a6d0479a91dab65cc24d5dfdfe99ac2e045611b/subsys/bluetooth/host/buf.c#L201-L204

Reply:
Ok. I see. That seems pretty hackish, though. The underscores in front of `__buf` is supposed to indicate that this is a private struct member, i.e. something that should not be touched outside of the net_buf implementation.

Reply:
Yeah. Initially we wanted to add the "view" mechanism to the net_buf API. But we thought it'd be better to stress-test it first in the Bluetooth subsys before attempting to "upstream" it so to speak.

If you think we should rather put the view API there directly I don't see any problem with that. We just didn't want to delay the PR more than what was necessary.

Reply:
I think it'd be better to do it in the net_buf subsystem, but the feature needs to at least:

 - be well defined
 - justify its existence in relation to e.g. net_buf_simple_state
 - cover all corner cases, e.g. what happens if you try to free a buffer whose data is heap allocated, but the __buf pointer has been modified through a view (the solution to this is probably as simple as the view holding a reference to the parent buffer, so that the buffer can't be destroyed before the view is)

I'm the maintainer for the net_buf subsystem, so I don't think there should be much risk of unnecessary delays for getting the feature in.